packages/devtools_app/lib/src/flutter/provider.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../auto_dispose.dart';
+import '../globals.dart';
+import '../logging/logging_controller.dart';
+
+/// Container for controllers that should outlive individual screens of the app.
+@immutable
+class ProviderData implements DisposableController {
```

Comment:
if this class is intended to be just for controllers, perhaps rename to ControllerProviderData. Same for below: r/Provider/ControllerProvider

Reply:
That's a good point, done.

We'll need to talk to these classes in a lot of places, so I've decided to go with a slightly shorter name.

We'll have `Controllers.of(context).logging` to get the logging controller, for instance.

Reply:
+100 on shorter names.